### ISSUE TITLE
In POSIX sh, readonly flags besides -p are undefined

### ISSUE THREAD

**Issue by contivero** [koalaman/shellcheck#1448]
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
foo() { echo bar; }
readonly -f foo
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
The only acceptable flag under POSIX sh for `readonly` is -p ([see spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#readonly)). The same is true for dash, but bash also accepts [-aAf]. I expected to see the usual "In POSIX sh, x flags are undefined".
